### PR TITLE
feat(EMS-3532): no pdf - application submission - XLSX - worksheets

### PR DESCRIPTION
**Pull request by ttbarnes**
## Introduction :pencil2:

This PR updates the XLSX generation so that each section of an application has it's on worksheet/tab in the XLSX, instead of single worksheet with a large list and a myriad of conditional index behaviours.

## Resolution :heavy_check_mark:

- Create new `INDEXES` constants for each section of an application.
- Create new `SECTION_NAMES` constants.
- Remove the original `XLSX_ROW_INDEXES` approach.
- Update `XLSX_HEADER_COLUMNS`.
- Update `generate.XLSX` to generate a worksheet for each section.
- Update `mapApplicationToXLSX` to return an object structure.
- Update `worksheetRowHeights` and `styledColumns` to work with worksheets instead of one single worksheet.

## Miscellaneous :heavy_plus_sign:

- Minor logging improvements.

### CODE REVIEW

File: src/api/constants/XLSX-CONFIG/INDEXES/EXPORTER_BUSINESS/index.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  ALTERNATIVE_TRADING_ADDRESS: 0,
+});
+
+/**
+ * EXPORTER_BUSINESS_INDEXES
+ * Generate row indexes for the XLSX's "Exporter business" worksheet.
+ * Depending on the submitted "Exporter business" data, new rows are rendered in the worksheet.
+ * - If "has different trading address" is true, the worksheet has 1 additional row.
+ * - If "has different trading name" is true, the worksheet has 1 additional row.
+ * @returns {Object}
```
Comment (Zainzzkk) [inline src/api/constants/XLSX-CONFIG/INDEXES/EXPORTER_BUSINESS/index.ts:19]: missing param

File: src/api/constants/XLSX-CONFIG/INDEXES/POLICY/BROKER_CONDITIONS/index.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+const { USING_BROKER } = POLICY_FIELD_IDS;
+
+/**
+ * BROKER_CONDITIONS
+ * Generate row indexes for the XLSX's "Policy" worksheet's "broker" conditions.
+ * Depending on the submitted "Policy" data, new rows are rendered in the worksheet.
+ * - If "different name on policy / policy contact is NOT the same as the owner", is true, the worksheet has 2 additional rows.
+ * - If "need pre-credit period cover" is true, the worksheet has 1 additional row.
+ * - If "requested a jointly insured party" is true, the worksheet has 3 additional rows.
+ * @returns {Object}
```
Comment (Zainzzkk) [inline src/api/constants/XLSX-CONFIG/INDEXES/POLICY/BROKER_CONDITIONS/index.ts:13]: same as above

File: src/api/constants/XLSX-CONFIG/INDEXES/POLICY/LOSS_PAYEE_CONDITIONS/index.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+/**
+ * LOSS_PAYEE_CONDITIONS
+ * Generate row indexes for the XLSX's "Policy" worksheet's "loss payee" conditions.
+ * Depending on the submitted "Policy" data, new rows are rendered in the worksheet.
+ * - If "different name on policy / policy contact is NOT the same as the owner", is true, the worksheet has 2 additional rows.
+ * - If "need pre-credit period cover" is true, the worksheet has 1 additional row.
+ * - If "requested a jointly insured party" is true, the worksheet has 3 additional rows.
+ * - If "using a broker" is true, the worksheet has 3 additional rows.
+ * @returns {Object}
```
Comment (Zainzzkk) [inline src/api/constants/XLSX-CONFIG/INDEXES/POLICY/LOSS_PAYEE_CONDITIONS/index.ts:14]: same as above

File: src/api/constants/XLSX-CONFIG/INDEXES/POLICY/index.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+ */
+export const DEFAULT_INDEXES = {
+  BROKER_ADDRESS: 0,
+  LOSS_PAYEE_ADDRESS: 0,
+};
+
+/**
+ * POLICY_INDEXES
+ * Generate row indexes for the XLSX's "Policy" worksheet.
+ * @returns {Object}
```
Comment (Zainzzkk) [inline src/api/constants/XLSX-CONFIG/INDEXES/POLICY/index.ts:24]: same as above

File: src/api/generate-xlsx/header-columns/index.ts [hunk 5]
```diff
@@ -7,9 +7,9 @@ const { KEY, VALUE, COLUMN_WIDTH } = XLSX_CONFIG;
  * Header columns for XLSX
  * @returns {Array} Columns for exceljs XLSX
  */
-const XLSX_HEADER_COLUMNS = [
-  { key: KEY.ID, header: KEY.COPY, width: COLUMN_WIDTH },
-  { key: VALUE.ID, header: VALUE.COPY, width: COLUMN_WIDTH },
+const XLSX_HEADER_COLUMNS = (sheetName: string) => [
+  { key: KEY, header: sheetName, width: COLUMN_WIDTH },
+  { key: VALUE, header: '', width: COLUMN_WIDTH },
```
Comment (Zainzzkk) [inline src/api/generate-xlsx/header-columns/index.ts:12]: does header need to be provided if empty string?

File: src/api/generate-xlsx/styled-columns/index.ts [hunk 6]
```diff
@@ -1,22 +1,32 @@
...
 const { LARGE_ADDITIONAL_COLUMN_HEIGHT, ADDITIONAL_TITLE_COLUMN_HEIGHT, FONT_SIZE } = XLSX_CONFIG;
 
+const { APPLICATION_INFORMATION } = SECTION_NAMES;
+
 /**
  * worksheetRowHeights
  * Add custom heights to certain worksheet cells
- * @param {Array} Title row indexes
- * @param {ExcelJS.Worksheet} ExcelJS worksheet
+ * @param {Array} rowIndexes: Row indexes
```
Comment (Zainzzkk) [inline src/api/generate-xlsx/styled-columns/index.ts:14]: Array<number>>

File: src/api/generate-xlsx/styled-columns/index.ts [hunk 7]
```diff
@@ -51,7 +55,13 @@ const styledColumns = (application: Application, worksheet: Worksheet) => {
...
       };
 
-      const isTitleRow = TITLE_INDEXES.includes(rowNumber);
+      const isInformationSheet = sheetName === APPLICATION_INFORMATION;
+      const isInformationTitleOne = isInformationSheet && rowNumber === 8;
+      const isInformationTitleTwo = isInformationSheet && rowNumber === 13;
+
+      const isInformationTitle = isInformationTitleOne || isInformationTitleTwo;
+
+      const isTitleRow = rowNumber === 1 || isInformationTitle;
```
Comment (ttbarnes) [inline src/api/generate-xlsx/styled-columns/index.ts:64]: Created a ticket to improve/extract the logic and add unit test coverage. EMS-3539

**Review by Zainzzkk — APPROVED** [review]

**Pull request merged**